modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/ValueProviderMediator.java

```diff
@@ -263,7 +263,9 @@ public ValueProviderMediator(T containerModel, Supplier<MuleContext> muleContext
         parameterModels.add(parameterModel);
       } else if (parameterModel.getFieldValueProviderModels()
           .stream()
-          .anyMatch(fieldValueProviderModel -> fieldValueProviderModel.getProviderName().equals(valueName))) {
+          .anyMatch(fieldValueProviderModel -> fieldValueProviderModel.getProviderName()
+              .substring(0, fieldValueProviderModel.getProviderName().lastIndexOf("_")).equals(valueName))) {
```

Comment:
I would externalize the logic of generating and handling providerNames with suffix so that it's together.